### PR TITLE
Unify hashing for avatar

### PR DESCRIPTION
**Pull request by Gusted**
- Unify the hashing code for repository and user avatars into a function.
- Use a sane hash function instead of MD5.
- Only require hashing once instead of twice(w.r.t. hashing for user avatar).
- Improve the comment for the hashing code of why it works.

### CODE REVIEW

File: modules/avatar/hash.go [hunk 1]
```diff
@@ -0,0 +1,26 @@
+package avatar
```
Comment (lunny) [inline modules/avatar/hash.go:4]: Copyright header missed.

Comment (yardenshoham) [reply modules/avatar/hash.go:4]:
```suggestion
// Copyright 2023 The Gitea Authors. All rights reserved.
// SPDX-License-Identifier: MIT

package avatar
```

**Review by delvh — APPROVED** [review]
We only change the URL where new avatars are stored with this PR, right?
So, existing avatars are unaffected?

**Comment by lunny** [comment]
`/drone/src/modules/avatar/hash.go:9:2: "fmt" imported but not used`

File: modules/avatar/hash.go [hunk 2]
```diff
@@ -0,0 +1,29 @@
+// Copyright 2023 The Gitea Authors. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+package avatar
+
+import (
+	"crypto/sha256"
+	"encoding/hex"
+	"fmt"
```
Comment (yardenshoham) [inline modules/avatar/hash.go]:
```suggestion
```

**Review by lunny — APPROVED** [review]

**Comment by delvh** [comment]
```go
--- FAIL: TestUploadAvatar (0.15s)

    avatar_test.go:31: 

        	Error Trace:	/drone/src/services/repository/avatar_test.go:31
        	Error:      	Not equal: 
        	            	expected: "10-9e24e19b024c44b778301d880bd8e6f4"
        	            	actual  : "d5fcd714cb1265ec33652f2d41c48b7f42a8fc1d282b8c577b3e997d206ab4b6"

        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -1 +1 @@
        	            	-10-9e24e19b024c44b778301d880bd8e6f4
        	            	+d5fcd714cb1265ec33652f2d41c48b7f42a8fc1d282b8c577b3e997d206ab4b6
        	Test:       	TestUploadAvatar

FAIL
```
I guess that failure makes sense?

**Comment by Gusted** [comment]
> We only change the URL where new avatars are stored with this PR, right? So, existing avatars are unaffected?

Yup that's correct.

**Pull request merged**